### ISSUE TITLE
SIGSEGV in `String.*`

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#49289]
```dart
void main() {
  print('abcd' * 0x4000000000000001);
}
```

```
===== CRASH =====
si_signo=Segmentation fault(11), si_code=2, si_addr=0x7fe8ced80000
version=2.18.0-edge.e78399ac52314e881f413aaf8975630f61e2e255 (be) (Sat Jun 18 18:17:40 2022 +0000) on "linux_x64"
pid=3975177, thread=3975185, isolate_group=main(0x5566a42a9000), isolate=main(0x5566a431d000)
isolate_instructions=5566a16c4300, vm_instructions=5566a16c4300
  pc 0x00007fe8ceda2f0a fp 0x00007fe8cd3fe298 Unknown symbol
  pc 0x00007fe8ceda20fe fp 0x00007fe8cd3fe2d0 Unknown symbol
  pc 0x00007fe8ceda2059 fp 0x00007fe8cd3fe2f8 Unknown symbol
  pc 0x00007fe8ceda1f8c fp 0x00007fe8cd3fe358 Unknown symbol
  pc 0x00007fe8ceda0e20 fp 0x00007fe8cd3fe390 Unknown symbol
  pc 0x00007fe8ceda0baf fp 0x00007fe8cd3fe3f8 Unknown symbol
  pc 0x00007fe8ceda0423 fp 0x00007fe8cd3fe448 Unknown symbol
  pc 0x00007fe8d5102a0c fp 0x00007fe8cd3fe4c0 Unknown symbol
  pc 0x00005566a18357dc fp 0x00007fe8cd3fe560 dart::DartEntry::InvokeCode(dart::Code const&, unsigned long, dart::Array const&, dart::Array const&, dart::Thread*)+0x14c
  pc 0x00005566a183562f fp 0x00007fe8cd3fe5c0 dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)+0x14f
  pc 0x00005566a1837a5c fp 0x00007fe8cd3fe610 dart::DartLibraryCalls::HandleMessage(long, dart::Instance const&)+0x14c
  pc 0x00005566a185eb4e fp 0x00007fe8cd3feba0 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message>>)+0x34e
  pc 0x00005566a188801d fp 0x00007fe8cd3fec10 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)+0x14d
  pc 0x00005566a18886fb fp 0x00007fe8cd3fec60 dart::MessageHandler::TaskCallback()+0x1db
  pc 0x00005566a19a82c8 fp 0x00007fe8cd3fece0 dart::ThreadPool::WorkerLoop(dart::ThreadPool::Worker*)+0x148
  pc 0x00005566a19a8728 fp 0x00007fe8cd3fed10 dart::ThreadPool::Worker::Main(unsigned long)+0x78
  pc 0x00005566a191deb8 fp 0x00007fe8cd3fedd0 out/ReleaseX64/dart-sdk/bin/dart+0x217feb8
-- End of DumpStackTrace
Aborted
```

**Comment by rakudrama** [comment]
The root cause is integer overflow in computing the size of the result, followed by unchecked writes.

Fix: https://dart-review.googlesource.com/c/sdk/+/249122

**Comment by rakudrama** [comment]
Fixed via 6dd17d6ec49e3a24c3344a8854850f58a0b18087
